databuilder/extractor/dashboard/tableau/tableau_dashboard_utils.py

```diff
@@ -0,0 +1,220 @@
+
+        return record
+
+
+class TableauDashboardAuth():
+    """
+    Attempts to authenticate agains the Tableau REST API using the provided personal access token credentials.
+    When successful, it will create a valid token that must be used on all subsequent requests.
+    https://help.tableau.com/current/api/rest_api/en-us/REST/rest_api_concepts_auth.htm
+    """
+
+    API_VERSION = const.API_VERSION
+    TABLEAU_HOST = const.TABLEAU_HOST
+    SITE_NAME = const.SITE_NAME
+    TABLEAU_ACCESS_TOKEN_NAME = const.TABLEAU_ACCESS_TOKEN_NAME
+    TABLEAU_ACCESS_TOKEN_SECRET = const.TABLEAU_ACCESS_TOKEN_SECRET
+    EXCLUDED_PROJECTS = const.EXCLUDED_PROJECTS
+    EXTERNAL_CLUSTER_NAME = const.EXTERNAL_CLUSTER_NAME
+    EXTERNAL_SCHEMA_NAME = const.EXTERNAL_SCHEMA_NAME
+    CLUSTER = const.CLUSTER
+    DATABASE = const.DATABASE
+
+    def __init__(self, conf):
+        self.site_id = None
+        self._token = None
+        self._conf = conf
+        self._site_name = self._conf.get_string(TableauDashboardAuth.SITE_NAME)
+        self._tableau_host = self._conf.get_string(TableauDashboardAuth.TABLEAU_HOST)
+        self._api_version = self._conf.get_string(TableauDashboardAuth.API_VERSION)
+        self._access_token_name = self._conf.get_string(TableauDashboardAuth.TABLEAU_ACCESS_TOKEN_NAME)
+        self._access_token_secret = self._conf.get_string(TableauDashboardAuth.TABLEAU_ACCESS_TOKEN_SECRET)
+
+    @property
+    def token(self):
+        if not self._token:
+            self._token = self._authenticate()
+        return self._token
+
+    def _authenticate(self):
+        """
+        Queries the auth/signin endpoint for the given Tableau instance using a personal access token.
+        The API version differs with your version of Tableau.
+        See https://help.tableau.com/current/api/rest_api/en-us/REST/rest_api_concepts_versions.htm
+        for details or ask your Tableau server administrator.
+        """
+        self._auth_url = "https://{tableau_host}/api/{api_version}/auth/signin".format(
+            tableau_host=self._tableau_host,
+            api_version=self._api_version
+        )
+        payload = json.dumps({
+            "credentials": {
+                "personalAccessTokenName": self._access_token_name,
+                "personalAccessTokenSecret": self._access_token_secret,
+                "site": {
+                    "contentUrl": self._site_name
+                }
+            }
+        })
+        headers = {
+            'Accept': 'application/json',
+            'Content-Type': 'application/json'
+        }
+        # verify = False is needed bypass occasional (valid) self-signed cert errors. TODO: actually fix it!!
+        params = {
+            "headers": headers,
+            "verify": False
+        }
+
+        response_json = requests.post(url=self._auth_url, data=payload, **params).json()
+        self.site_id = response_json['credentials']['site']['id']
```

Comment:
Is `self.site_id` used anywhere?

Reply:
Nope. It was relic from when I was using the REST API during the hackathon, it's not needed for the Metadata API so I'll nix it